src/main/java/org/springframework/data/neo4j/core/Neo4jOperations.java

```diff
@@ -191,6 +193,24 @@
 	 */
 	<T> T save(T instance);
 
+	/**
+	 * Saves an instance of an entity, using the provided predicate to shape the stored graph. One can think of the predicate
+	 * as a dynamic projection. If you want to save or update properties of associations (aka related nodes), you must include
+	 * the association property as well (meaning the predicate must return {@literal true} for that property, too).
+	 * <p>
+	 * Be careful when reusing the returned instance for further persistence operations, as it will most likely not be
+	 * fully hydrated and without using a static or dynamic projection, you will most likely cause data loss.
+	 *
+	 * @param instance        the entity to be saved. Must not be {@code null}.
+	 * @param includeProperty A predicate to determine the properties to save.
+	 * @param <T>             the type of the entity.
+	 * @return the saved instance.
+	 * @since 6.3
+	 */
+	default <T> T saveAs(T instance, BiPredicate<PropertyPath, Neo4jPersistentProperty> includeProperty) {
```

Comment:
Since we accept it, the instance should be `@Nullable`. Would also make sense for the `saveAllAs` method and the reactive counter-parts. (Would need an additional null check in the `saveAllAs`/`saveAllImpl`).